packages/react-badge/src/components/CounterBadge/useCounterBadgeStyles.ts

```diff
@@ -49,5 +49,5 @@ export const useCounterBadgeStyles = (state: CounterBadgeState) => {
     !state.showZero && state.count === 0 && !state.dot && styles.hide,
     state.className,
   );
-  return useBadgeStyles(state);
+  return useBadgeStyles(state) as CounterBadgeState;
```

Comment:
Why doesn't the returned value match the `CounterBadgeState` type without the cast? Seems like it might hide a real issue. (same in next file)

Reply:
```
src/components/CounterBadge/useCounterBadgeStyles.ts:52:5 - error TS2739: Type 'BadgeState' is missing the following properties from type 'CounterBadgeState': overflowCount, count, showZero, dot
```

Looking at the implementation of `useBadgeStyles`, it can (and should be) updated to reflect what it actually returns, which is mostly what was passed in. I'm happy to leave a TODO or open an issue, but I'd rather not modify `useBadgeStyles` at this point in the PR. Thoughts on the best next step?

Reply:
Agreed we probably shouldn't change it here. Maybe open an issue and assign to the component author? (forgetting who that is)